### ISSUE TITLE
password is stored in cleartext

### ISSUE THREAD

**Issue by jonassmedegaard** [joepio/atomic-data-rust#370]
`atomic-server` stores password in cleartext in its config file.

That may be acceptable (or even needed) for some scenarios, but please support secure storage - and preferably by default and documented as the recommended way.

Seems [keyring](https://crates.io/crates/keyring) is suitable for this.

**Comment by joepio** [comment]
Yeah it's definitely a weak spot, although I'm not entirely sure how I can provide enough convenience with a tool like keyring.

Users should be able to read their private key in an easy way. If the key itself is stored with `keyring`, where can users find it? Maybe I'll need to add commands to both `atomic-cli` and `atomic-server`? Or is there a different way?

**Comment by jonassmedegaard** [comment]
I don't know `keyring`, only the XDG protocol used for the linux backend - and there normal user experience has an independent approval interface pop up when your tool wants to fetch a secret.  I.e. it is optimized for use as a desktop-oriented daemon.

Perhaps it is helpful to discuss issues like this one, if you document the types of scenarios you want to support with the daemon and CLI tool.

As I understand it, you want to support at least three scenarios:
 1. self-contained system daemon - network-accessible with auto-created ACME certificate
 2. backend system service - accessible only from a frontend web service, either without using the ACME feature or optionally compiled without support for it at all
 3. self-contained desktop service - run by a local users and accessible only at localhost (unencrypted, assuming that user is the only one on that host)

As I see it, at least scenario 3) should use `keyring` but some might also want their system services to store secrets on special devices, e.g. on an external USB-connected storage that needs to be plugged in when services are started, and then is unplugged again - expecting secrets to only ever reside in protected memory blocks.
